src/inventory.h

```diff
@@ -232,28 +235,78 @@ class InventoryList
 	// If not as many items exist as requested, removes as
 	// many as possible.
 	// Returns the items that were actually removed.
-	ItemStack removeItem(const ItemStack &item);
+	ItemStack removeItem(bool script_callback, const ItemStack &item);
 
 	// Takes some items from a slot.
 	// If there are not enough, takes as many as it can.
 	// Returns empty item if couldn't take any.
-	ItemStack takeItem(u32 i, u32 takecount);
+	ItemStack takeItem(bool script_callback, u32 i, u32 takecount);
 
 	// Move an item to a different list (or a different stack in the same list)
 	// count is the maximum number of items to move (0 for everything)
 	// returns number of moved items
-	u32 moveItem(u32 i, InventoryList *dest, u32 dest_i,
+	u32 moveItem(bool script_callback, u32 i, InventoryList *dest, u32 dest_i,
 		u32 count = 0, bool swap_if_needed = true, bool *did_swap = NULL);
 
 	// like moveItem, but without a fixed destination index
 	// also with optional rollback recording
-	void moveItemSomewhere(u32 i, InventoryList *dest, u32 count);
+	void moveItemSomewhere(bool script_callback, u32 i, InventoryList *dest, u32 count);
 
 private:
 	std::vector<ItemStack> m_items;
 	std::string m_name;
 	u32 m_size, m_width;
 	IItemDefManager *m_itemdef;
+	InventoryChangeReceiver *m_rec;
+};
+
+class InventoryChangeReceiver
+{
+public:
+	InventoryChangeReceiver() { }
+	InventoryChangeReceiver(const Scripting script)
+	{
+		m_script = script;
```

Comment:
copying should be disabled for Scripting, anyway (ie: this shouldn't even compile)